rita/src/rita_client/dashboard/interfaces.rs

```diff
@@ -0,0 +1,518 @@
+
+    match a {
+        InterfaceMode::WAN => unimplemented!(),
+        // nothing to do here we overwrite everything we need later
+        InterfaceMode::LAN => {}
+        // for mesh we need to send an unlisten and delete the static interface we made
+        InterfaceMode::Mesh => {
+            let ret = KI.del_uci_var(&format!("network.rita_{}", ifname));
+            return_codes.push(ret);
+            PeerListener::from_registry().do_send(UnListen(ifname.clone().to_string()));
+        }
+        InterfaceMode::Unknown => unimplemented!(),
+    }
+
+    match b {
+        InterfaceMode::WAN => unimplemented!(),
+        // since we left lan mostly unomidifed we just pop in the ifname
+        InterfaceMode::LAN => {
+            let ret = KI.set_uci_var(&format!("wireless.{}.network", network_section), "lan");
+            return_codes.push(ret);
+            let ret = KI.set_uci_var(&format!("wireless.{}.mode", network_section), "ap");
+            return_codes.push(ret);
+            let ret = KI.set_uci_var(&format!("wireless.{}.ssid", network_section), "AltheaHome");
+            return_codes.push(ret);
+            let ret = KI.set_uci_var(
+                &format!("wireless.{}.encryption", network_section),
+                "psk2+tkip+aes",
+            );
+            return_codes.push(ret);
+            let ret = KI.set_uci_var(&format!("wireless.{}.key", network_section), "ChangeMe");
+            return_codes.push(ret);
+        }
+        // in this section we modfiy the wlan config to mesh and then add a static logical iface
+        // that is used for things like ip assignment etc
+        InterfaceMode::Mesh => {
+            let ret = KI.set_uci_var(
+                &format!("wireless.{}.network", network_section),
+                &format!("rita_{}", ifname),
+            );
+            return_codes.push(ret);
+            // TODO detect if the driver perfers adhoc or meshpoint
+            let ret = KI.set_uci_var(&format!("wireless.{}.mode", network_section), "adhoc");
+            return_codes.push(ret);
+            let ret = KI.set_uci_var(&format!("wireless.{}.ssid", network_section), "AltheaMesh");
+            return_codes.push(ret);
+            let ret = KI.set_uci_var(&format!("wireless.{}.encryption", network_section), "none");
+            return_codes.push(ret);
+
+            let ret = KI.set_uci_var(&format!("network.rita_{}", ifname), "interface");
+            return_codes.push(ret);
+            let ret = KI.set_uci_var(&format!("network.rita_{}.ifname", ifname), ifname);
+            return_codes.push(ret);
+            let ret = KI.set_uci_var(&format!("network.rita_{}.proto", ifname), "static");
+            return_codes.push(ret);
+            mesh_add = true;
+        }
+        InterfaceMode::Unknown => unimplemented!(),
+    }
+
+    // check all of our return codes in order to handle any possible issue
+    let mut error_occured = false;
+    for ret in return_codes {
```

Comment:
I think it could be rewritten as something like `return_codes.map(|r| r.is_ok()).any()` or something similiar.

Reply:
sure it could, but it would be harder to read and no more efficient.